tornado-unittests/src/main/java/uk/ac/manchester/tornado/unittests/prebuilt/PrebuiltTests.java

```diff
@@ -342,4 +347,83 @@ public void testPrebuilt04SPIRVThroughOpenCLRuntime() throws TornadoExecutionPla
         assertEquals(512, finalSum, 0.0f);
 
     }
+
+    /**
+     * This test is intended to be passed with multiple backends (e.g., OpenCL and PTX).
+     * The PTX backend needs to be installed. Otherwise, an exception is thrown.
+     *
+     * <p> How to run?
+     * <code>
+     * tornado-test -V uk.ac.manchester.tornado.unittests.prebuilt.PrebuiltTests#testPrebuiltMutiBackend
+     * </code>
+     * </p>
+     * 
+     * @throws TornadoExecutionPlanException
+     */
+    @Test
+    public void testPrebuiltMutiBackend() throws TornadoExecutionPlanException {
+
+        final int numElements = 8;
+        IntArray a = new IntArray(numElements);
+        IntArray b = new IntArray(numElements);
+        IntArray c = new IntArray(numElements);
+
+        a.init(1);
+        b.init(2);
+
+        // Force to use the PTX Backend.
+        FILE_PATH += "add.ptx";
+
+        TornadoDeviceMap tornadoDeviceMap = TornadoExecutionPlan.getTornadoDeviceMap();
+        if (tornadoDeviceMap.getNumBackends() < 2) {
+            throw new TornadoVMMultiDeviceNotSupported("Test designed to run with multiple backends");
+        }
+
+        List<TornadoBackend> ptxBackend = tornadoDeviceMap.getBackendsWithPredicate(backend -> backend.getBackendType() == TornadoVMBackendType.PTX);
+
+        if (ptxBackend == null || ptxBackend.isEmpty()) {
+            throw new TornadoVMPTXNotSupported("Test designed to run with multiple backends, including a PTX backend");
+        }
```

Comment:
I am not very keen to have the check in the test body. Shall we change to something like the following condition 
```java
assertAvailableDrivers(2);
```
that exists in `TestConcurrentBackends.java` class.

The condition should assert if (OpenCL, PTX) or (SPIR-V, PTX) backends have been built, otherwise result in UNSUPPORTED result.

Reply:
We need to check that at least two drivers are installed, and one of them must be the PTX to pass this particular test. The combinations are also valid for any two backend combination in which the selected device is not the default device. I think the throw exception here is fine.